pkg/pipeline/transform/connection_tracking/connection_tracking_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2021 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package connection_tracking
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func Test_InitConnectionTracking(t *testing.T) {
+	InitConnectionTracking()
+	require.NotNil(t, CT)
+}
+
+func Test_AddFlow(t *testing.T) {
+	InitConnectionTracking()
+	isNew := CT.AddFlow("test")
+	require.Equal(t, true, isNew)
+	require.Equal(t, CT.cacheMap[getHashKey("test")].flowlogsCount, 1)
+
+	isNew = CT.AddFlow("test")
+	require.Equal(t, false, isNew)
+	require.Equal(t, CT.cacheMap[getHashKey("test")].flowlogsCount, 2)
+}
+
+func Test_IsFlowNew(t *testing.T) {
+	InitConnectionTracking()
+	_ = CT.AddFlow("test")
+	isNew := CT.IsFlowNew("test")
+	require.Equal(t, true, isNew)
+	isNew = CT.IsFlowNew("test_false")
```

Comment:
The name `IsFlowNew` is confusing. The "test_false" flow will not be in the cache so it will return `false`. This is not because it is not new, but rather because it does not exist. An entry is in the cache not because it is new, but rather because it is currently active.

Reply:
Renamed to `IsFlowKnown`